### PR TITLE
CMS-473: Add no results messaging

### PR DESCRIPTION
**Pull request by ayumi-oxd**
### Jira Ticket:
CMS-473

### Description:
- Update no results messaging
- Add no results messaging to the dates page, A-Z list page, and the approved page

### CODE REVIEW

File: src/gatsby/src/components/search/noSearchResults.js [hunk 1]
```diff
@@ -2,93 +2,133 @@ import React from "react"
...
-              aria-label="Remove city name"
-              onClick={handleClickClearCity}
-              onKeyDown={handleKeyDownClearCity}
+      {page === "find-a-park" && (
+        <>
+          <p><b>You could:</b></p>
+          <ul>
+            <li>Try a different search term</li>
+            <li>
+              <button
```
Comment (duncan-oxd) [inline src/gatsby/src/components/search/noSearchResults.js:68]: You can "press" a button with a mouse click, or by focusing it and pressing space (or enter) but I _think_ all 3 of these event swill fire a `click` event, so you don't need to specifically handle onKeyDown. (Double check if that's true in React, though...)

File: src/gatsby/src/components/search/noSearchResults.js [hunk 2]
```diff
@@ -2,93 +2,133 @@ import React from "react"
...
+      )}
+      {(page === "park-operating-dates" || page === "a-z-list") && (
+        <>
+          <p className="mb-4">
+            <b>Select 'All' to go back to the full list.</b>
+          </p>
+          <p><b>Or search using the:</b></p>
+          <ul>
+            <li>
+              <Link to="/find-a-park">Find a park</Link> page
```
Comment (duncan-oxd) [inline src/gatsby/src/components/search/noSearchResults.js:98]: I remember we updated the links in #1531 to include "page" in the anchor text; do we want to do that here too? I see "...list" is already included in the A-Z list link.

File: src/gatsby/src/components/search/noSearchResults.js (hunk 1 above)
Comment (ayumi-oxd) [reply src/gatsby/src/components/search/noSearchResults.js:62]: Good catch! Yes it works in React as well.

File: src/gatsby/src/components/search/noSearchResults.js (hunk 2 above)
Comment (ayumi-oxd) [reply src/gatsby/src/components/search/noSearchResults.js]: Thanks! Brett prefers to include "page" as a link.

**Review by duncan-oxd — APPROVED** [review]

**Pull request merged**